pyout/tests/test_tabular.py

```diff
@@ -852,6 +853,32 @@ def test_tabular_write_callable_values_multicol_key_infer_column(result):
     assert len([ln for ln in lines if ln.endswith("foo done /tmp/a")]) == 1
 
 
+def delayed_updates():
+    for val in ["update", "finished"]:
+        time.sleep(0.05)
+        yield val
+
+
+@pytest.mark.timeout(10)
+@pytest.mark.parametrize("gen_source",
+                         [delayed_updates, delayed_updates()],
+                         ids=["gen_func", "generator"])
+@patch("pyout.tabular.Terminal", TestTerminal)
+def test_tabular_write_generator_function_values(gen_source):
+    fd = StringIO()
+    with Tabular(["name", "status"], stream=fd) as out:
+        out({"name": "foo", "status": ("waiting", gen_source)})
+        out({"name": "bar", "status": "ok"})
+
+        expected = ("foo waiting\n"
+                    "bar ok     \n")
+        assert eq_repr(fd.getvalue(), expected)
+    lines = fd.getvalue().splitlines()
+    assert len([ln for ln in lines if ln.endswith("foo update ")]) == 1
+    assert len([ln for ln in lines if ln.endswith("foo finished")]) == 1
+    assert len([ln for ln in lines if ln.endswith("bar ok      ")]) == 1
```

Comment:
Unrelated to this actual PR but related to [this discussion][0] about
the need for more test fixtures/helpers: yes, this
`assert len(... == 1` pattern is repeated all over the tests and
should be abstracted out into a helper.


[0]: https://github.com/pyout/pyout/pull/33#discussion_r165061634

Reply:
+1